### PR TITLE
fix: Pipeline agent image update from 18.04 to 22.04 | AzSecPack

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

This PR is the first to migrate our agent pool to a 1ES maintained agent pool. This is primarily to unblock the ACN PR pipeline and will be improved upon. 

Future CI:
- Check for systemd-resolved not tied to ubuntu image version
- Properly assign *auth.json to skopeo commands
- Remove reinstalling dependencies from JIT provisioning script
- Remove all uses of `sudo` where applicable

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by estebancams — APPROVED** [review]

**Comment by jpayne3506** [comment]
Will integrate changes from https://github.com/Azure/azure-container-networking/pull/1985 which https://github.com/Azure/azure-container-networking/pull/2320#issuecomment-1777556623 mentions in the matching closed PR

File: Makefile [hunk 1]
```diff
@@ -499,7 +499,7 @@ manifest-push: # util target to push multiarch container manifest.
 	$(CONTAINER_BUILDER) manifest push --all $(IMAGE_REGISTRY)/$(IMAGE):$(TAG) docker://$(IMAGE_REGISTRY)/$(IMAGE):$(TAG)
 
 manifest-skopeo-archive: # util target to export tar archive of multiarch container manifest.
-	skopeo copy --all docker://$(IMAGE_REGISTRY)/$(IMAGE):$(TAG) oci-archive:$(IMAGE_ARCHIVE_DIR)/$(IMAGE)-$(TAG).tar
+	skopeo copy --all docker://$(IMAGE_REGISTRY)/$(IMAGE):$(TAG) oci-archive:$(IMAGE_ARCHIVE_DIR)/$(IMAGE)-$(TAG).tar --debug
```
Comment (sheylatrudo) [inline Makefile:502]: Do we want to keep this as debug?

**Review by sheylatrudo — APPROVED** [review]
Looks good to me!

File: .pipelines/containers/manifest-template.yaml [hunk 2]
```diff
@@ -19,9 +19,16 @@ steps:
...
-      set -e
+      set -ex
+      echo "checking XDG_RUNTIME_DIR"
+      echo $XDG_RUNTIME_DIR
       make ${{ parameters.name }}-manifest-push
       mkdir -p $(Build.ArtifactStagingDirectory)/images
+
+      echo "setting XDG_RUNTIME_DIR"
+      export XDG_RUNTIME_DIR=/run/user/$(id -u)
+      echo $XDG_RUNTIME_DIR
```
Comment (sheylatrudo) [inline .pipelines/containers/manifest-template.yaml:30]: Do we want to set this in the task or set this as an env var from the pipeline level?

Comment (jpayne3506) [reply .pipelines/containers/manifest-template.yaml:30]: If we set this as an env var in the pipeline then it impacts podman and buildah negatively. This hack is exclusive to skopeo.

File: Makefile (hunk 1 above)
Comment (jpayne3506) [reply Makefile:502]: I believe we should as it provides good understanding of what is happening with the `*auth.json` files IRT skopeo. Future changes to skopeo could break the current hack and future CI changes of this command.

**Review by paulyufan2 — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**